### ISSUE TITLE
Unable to connect after installing readline-common

### ISSUE THREAD

**Issue by feryllt** [CypherpunkArmory/UserLAnd#161]
Hi all,
I am unable to connect to a debian filesystem after installing `readline-common`. ConnectBot refuses connection, after asking for password, saying "connection lost".

Steps to reproduce:

1. Create debian filesystem
2. Create ssh connection
3. Connect using ConnectBot (tried also with JuiceSSH)
4. Run `sudo apt update && sudo apt install -y readline-common && exit`
5. Kill service in UserLAnd
6. Connect again, insert password
7. Fails here with "Connection lost"

Configuration:
Huawei MediaPad M5 10.8 official image Android Oreo CMR-W09 8.0.0.161 (C432) kernel 4.4.23 aarch64

Please ask for any info/log/experiment needed.

**Comment by emidiomorgia** [comment]
Hello, I have the same issue after installing lxde, xfce or even any simple apt-get related package. I couldn't never try a gui desktop environment because after first apt-get install <something>, the connection was lost forever and any subsequent SSH connection never goes. I always receive the message:  'connection lost'. Rebooting the device does not solve the problem.

Thank you in advance

**Comment by corbinlc** [comment]
@feryllt Can you grab a PRoot log? You do this by stopping your session if it is running currently. Then go to settings inside the UserLAnd app and turn on PRoot debugging and setting the verbosity to 9. Then you launch your sessions again and run the command that is giving you trouble (in your case just trying to connect will faile). Then you should find a file on your sdcard (real or emulated) called PRoot_Debug_Log. Please share that with us one way or another as it will help debug this.  

Sounds like @luongthomas has already been able to replicate this issue.  So, that is good.  I don't have the Huawei device with me until next week and this seems to work fine on my test device.  But, a log would help.

**Comment by corbinlc** [comment]
@emidiomorgia Can you provide what device and android version you are using?  Can you also provide a proot log of trying to connect after some basic package install (please tell us what package you installed, so we can try to replicate).

**Comment by emidiomorgia** [comment]
The device is Huawei mate 10 pro and android version is 8.0.0The log with verbose level 9 is in attachment. I set up Debian fs with connectbot. Once started userland I did a sudo apt update, after updated I did a sudo apt-get install lxde. While installing the connection was lost and I connected again and finished installation with dpkg - - configure - aThe setup finished and I had prompt ready, as I disconnected and tried to reconnect, connectbot always say connection lost after entering the password. Let me know if you need more info. Regards EmidioIl 11 ago 2018 08:31, corbinlc <notifications@github.com> ha scritto:@emidiomorgia Can you provide what device and android version you are using?  Can you also provide a proot log of trying to connect after some basic package install (please tell us what package you installed, so we can try to replicate).

**Comment by feryllt** [comment]
@corbinlc Please find the log here [PRoot_Debug_Log.log](https://github.com/CypherpunkArmory/UserLAnd/files/2280284/PRoot_Debug_Log.log).

What I have done:

1. Create debian fs
2. Create ssh connection
3. Launch service
4. Connect and execute `sudo apt update && sudo apt install -y readline-common && exit`
5. Kill service in UserLAnd
6. Delete previous log file
7. Enable logging at level 9
8. Launch service
9. Connect
10. Get connection failure as explained in issue description
11. Kill server
12. Disable logging

**Comment by emidiomorgia** [comment]
Hello, connection lost persists even after the last update and using connectbot or juice SSH. I removed and installed again several times but the error persists even if I use vnc session the terminal window disappears. Since from the first start the terminal window in vnc session is Chrome less without title and borders and the only window manager I can install without loosing connection is xfwm but after installed any window I open is Chromeless and disappears immediately. The device is Huawei mate 10 pro and Android version is 8.0.0

**Comment by corbinlc** [comment]
There are two issues described here, but I think both are working with the latest release.  @ferylit can you confirm if the flow described now works on a new filesystem (not sure if a broken one would be fixed).  @emidiomorgia can you describe what you see now?

**Comment by emidiomorgia** [comment]
@corbinlc I removed userland and any vnc client. I installed again userland and created fs of type Debian. I created a vnc session and once started I can only see a white console with no border or title on window: it is Chromeless. So I made the following steps after updating to latest packages: sudo apt-get install xfce4. After few packages the system asked me what keyboard to use and I choose Italian (53) after few steps the connection is lost this time with an error from vnc client telling me that connection is lost and showing me the Java stack trace. After that userland has the service running but I can't connect anymore. If I kill the service and start session again I can't use apt-get because of corrupted packages and I have to use sudo dpkg - - configure -a then the system start again installing but it stops few seconds later with the same error. This king of error is different from previous one where the connection was lost permanently but the strange is the console window without borders and title and I can't go on with installation anymore because of the refuted connection. Android version is 8 and the device is Huawei mate 10 pro.
